File: 백준/Silver/1004. 어린 왕자/어린 왕자.py
import sys
input = sys.stdin.readline

tc = int(input())
n = 0
points = []
x1, y1, x2, y2 = 0, 0, 0, 0


def dist_square(y1, x1, y2, x2):
    return ((y2 - y1) ** 2 + (x2 - x1) ** 2)


def main():
    cnt = 0
    for y, x, r in points:
        a = dist_square(y, x, y1, x1) <= r ** 2
        b = dist_square(y, x, y2, x2) <= r ** 2
        if a ^ b:
            cnt += 1

    return cnt


for _ in range(tc):
    x1, y1, x2, y2 = map(int, input().split())
    n = int(input())
    points = []
    for _ in range(n):
        x, y, r = map(int, input().split())
        points.append((y, x, r))

    print(main())

File: 백준/Silver/1004. 어린 왕자/어린 왕자.js
const fs = require('fs');
const stdin = fs.readFileSync('dev/stdin').toString().trim();

const read = stdin.split("\n");
const tc = Number(read[0]);
let rest = read.slice(1);

for(let i = 0; i < tc; i ++ ){
    let [x1, y1, x2, y2] = rest[0].split(" ").map(Number);
    let n = Number(rest[1]);    
    
    let points = rest.slice(2, 2 + n).map(p => p.split(" ").map(Number));
    rest = rest.slice(2 + n);
    
    const dist = (x1, y1, x2, y2) => {
        return (y1 - y2) ** 2 + (x1 - x2) ** 2;
    }
    let cnt = 0;
    for(let [x,y,r] of points){
        let a = dist(x, y, x1, y1) <= r ** 2;
        let b = dist(x, y, x2, y2) <= r ** 2;
        if((a || b) && a != b) {
            cnt += 1;
        }
    }
    
    console.log(cnt);
}